android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/ssdp/Ssdp.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2014 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.physical_web.physicalweb.ssdp;
+
+import android.util.Log;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+
+/**
+ * This class implements SSDP protocol.  
+ * It supports discovery of SSDP devices in
+ * the local network using SSDP M-SEARCH search requests
+ */
+public class Ssdp implements Runnable {
+    public static final String TAG = "Ssdp";
+    public static final String SSDP_ADDRESS = "239.255.255.250";
+    public static final int SSDP_PORT = 1900;
+    public static final String SSDP_HOST = SSDP_ADDRESS + ":" + SSDP_PORT;
+    public static final String MAX_AGE = "max-age=1800";
+    public static final int TTL = 128;
+    public static final int MX = 2;
+    public static final String ALIVE = "ssdp:alive";
+    public static final String BYEBYE = "ssdp:byebye";
+    public static final String UPDATE = "ssdp:update";
+    public static final String TYPE_M_SEARCH = "M-SEARCH";
+    public static final String TYPE_NOTIFY = "NOTIFY";
+    public static final String TYPE_200_OK = "200 OK";
+
+    private SsdpCallback mSsdpCallback;
+    private SocketAddress mMulticastGroup;
+    private DatagramSocket mDatagramSocket;
+    private Thread mThread;
+    public Ssdp(SsdpCallback ssdpCallback) throws IOException{
+        mSsdpCallback = ssdpCallback;
+        mMulticastGroup = new InetSocketAddress(SSDP_ADDRESS,SSDP_PORT);
```

Comment:
Please separate the arguments with a space after each comma (e.g. mMulticastGroup = new InetSocketAddress(SSDP_ADDRESS, SSDP_PORT);